jujugui/static/gui/src/app/models/models.js

```diff
@@ -2537,6 +2537,22 @@ YUI.add('juju-models', function(Y) {
     },
 
     /**
+      Update the map for a unit id on a service.
+
+      @method updateUnitMap
+      @param {Model} service The service model.
+      @param {String} oldId The previous unit id.
+      @param {Model} newModel The new unit model.
+    */
+    updateUnitMap: function(service, oldId, newModel) {
```

Comment:
I'd move the db.units modifications into the updateUnitMap method and call it updateUnitId so that it follows the same pattern as the add/removeUnits methods